connector/avro/src/test/scala/org/apache/spark/sql/avro/AvroCodecSuite.scala

```diff
@@ -27,19 +30,34 @@ class AvroCodecSuite extends FileSourceCodecSuite {
   override protected def availableCodecs =
     AvroCompressionCodec.values().map(_.lowerCaseName()).iterator.to(Seq)
 
-  availableCodecs.foreach { codec =>
+  (availableCodecs ++ availableCodecs.map(_.capitalize)).foreach { codec =>
     test(s"SPARK-46746: attach codec name to avro files - codec $codec") {
       withTable("avro_t") {
         sql(
           s"""CREATE TABLE avro_t
              | USING $format OPTIONS('compression'='$codec')
              | AS SELECT 1 as id
              | """.stripMargin)
-        spark.table("avro_t")
-          .inputFiles.foreach { file =>
-            assert(file.endsWith(s"$codec.avro".stripPrefix("uncompressed")))
+        spark
+          .table("avro_t")
+          .inputFiles.foreach { f =>
+            assert(f.endsWith(s"$codec.avro".toLowerCase(Locale.ROOT).stripPrefix("uncompressed")))
           }
       }
     }
   }
+
+  test("SPARK-46754: invalid compression codec name in avro table definition") {
+    checkError(
+      exception = intercept[SparkIllegalArgumentException](
+        sql(
+          s"""CREATE TABLE avro_t
+             | USING $format OPTIONS('compression'='unsupported')
```

Comment:
Although this is a copy of the existing test case pattern, shall we remove the space?
```scala
- | USING $format OPTIONS('compression'='unsupported')
+ |USING $format OPTIONS('compression'='unsupported')
```

Reply:
I have removed any extra spaces and blank lines from both the origin and this one. Thank you for pointing this out